app/src/androidTest/java/com/github/sdp_begreen/begreen/activities/MainActivityTest.kt

```diff
@@ -416,4 +422,64 @@ class MainActivityTest {
             }
         }
     }
+
+    @Test
+    fun clickOnCapturePhotoDontThrowsError() {
+
+        activityRule.scenario.onActivity {
+            val connectedUserViewModel by it.viewModels<ConnectedUserViewModel>()
+            connectedUserViewModel.setCurrentUser(user1)
+        }
+        onView(withId(R.id.camera_capture_button)).perform(click())
+        //BaseRobot().assertOnView(withId(R.id.sendPostFragment), matches(isDisplayed()))
+
+    }
+
+    @Test
+    fun clickOnProfileDetailsCameraRedirectCorrectly() {
+
+        activityRule.scenario.onActivity {
+            val connectedUserViewModel by it.viewModels<ConnectedUserViewModel>()
+            connectedUserViewModel.setCurrentUser(user1)
+        }
+        onView(withId(R.id.profile_cam)).perform(click())
+        BaseRobot().assertOnView(withId(R.id.fragment_profile_details), matches(isDisplayed()))
+
+    }
+
+    @Test
+    fun clickOnSendPostRedirectToPreviewCorrectly() {
+
+        activityRule.scenario.onActivity {
+            val connectedUserViewModel by it.viewModels<ConnectedUserViewModel>()
+            connectedUserViewModel.setCurrentUser(user1)
+
+
+
+            it.supportFragmentManager.beginTransaction()
```

Comment:
I am not sure I understand it correctly, but here, don't you change the fragment manually, and then "simulate" a click on the button ?

Is it the trick that you told me about to make the test work ?